lldb/source/Plugins/ObjectFile/Minidump/MinidumpFileBuilder.cpp

```diff
@@ -686,16 +692,10 @@ Status MinidumpFileBuilder::AddExceptions() {
   for (const ThreadSP &thread_sp : thread_list) {
     StopInfoSP stop_info_sp = thread_sp->GetStopInfo();
     bool add_exception = false;
-    if (stop_info_sp) {
-      switch (stop_info_sp->GetStopReason()) {
-      case eStopReasonSignal:
-      case eStopReasonException:
+    if (stop_info_sp && thread_stop_reasons.count(stop_info_sp->GetStopReason()) > 0) {
         add_exception = true;
-        break;
-      default:
-        break;
-      }
     }
+
```

Comment:
Down below we should do something like:
```
const StopReason stop_reason = stop_info_sp->GetStopReason();
if (stop_reason == eStopReasonSignal || stop_reason == eStopReasonException)
  exp_record.ExceptionCode = static_cast<llvm::support::ulittle32_t>(stop_info_sp->GetValue());
else
  exp_record.ExceptionCode = lldb::StopReason::eStopReasonLLDB;
```
Then fill in the arguments for each stop reason into:
```
  size_t data_count = ...; // See SBThread::GetStopReasonDataCount()
  exp_record.NumberParameters = data_count; // See SBThread::GetStopReasonDataCount()
  for (i=0; i<data_count; ++i)
    exp_record.ExceptionInformation[i] = ...; // See SBThread::GetStopReasonDataAtIndex()
```
Then we learn to load this information into a StopReason using this data. Then we never lose any signal.

Reply:
https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-exception_record

The ExceptionCode should be defined it terms of `EXCEPTION_ACCESS_VIOLATION` and `EXCEPTION_BREAKPOINT` right? We can make up new ExceptionCode values for LLDB, but they shouldn't interfere with the existing ones. That is unless each OS can redefine these ExceptionCode values, which might be the case. I would check the breakpad codebase to see what they do since that is the other major source of minidump files we consume (breakpad created ones).

Reply:
So I ended up just slapping the description in there and setting the flags with an LLDB flag, thoughts?